static/js/careers-filter-and-sort.js

```diff
@@ -75,16 +155,38 @@
   function filterJobs(filterBy, jobList) {
     numberOfJobsDisplayed = domList.childElementCount;
     jobList.forEach(function (node) {
-      if (filterBy.filterText === "All") {
+      if (filterBy.filterText === "All" && filterBy.location === "all") {
         if (node.classList.contains("u-hide")) {
           node.classList.remove("u-hide");
         }
         numberOfJobsDisplayed = domList.childElementCount;
-      } else {
+      } else if (filterBy.filterValue !== "all" && filterBy.location === "all") {
         if (node.dataset[filterBy.filterName].includes(filterBy.filterText)) {
           if (node.classList.contains("u-hide")) {
             node.classList.remove("u-hide");
           }
+          } else {
+            if (!node.classList.contains("u-hide")) {
+              node.classList.add("u-hide");
+          }
+          numberOfJobsDisplayed--;
+        }
+      } else if (filterBy.filterValue === "all" && filterBy.location !== "all") {
+        if (node.getAttribute("location-filter").includes(filterBy.location)) {
+          if (node.classList.contains("u-hide")) {
+            node.classList.remove("u-hide");
+          }
+          } else {
+            if (!node.classList.contains("u-hide")) {
+              node.classList.add("u-hide");
+          }
+          numberOfJobsDisplayed--;
+        }
```

Comment:
Small indentation tweak here - at first glance this looked like a verbose way of writing `node.classList.toggle("u-hide")`
```suggestion
        } else {
          if (!node.classList.contains("u-hide")) {
            node.classList.add("u-hide");
          }
          numberOfJobsDisplayed--;
        }
      } else if (filterBy.filterValue === "all" && filterBy.location !== "all") {
        if (node.getAttribute("location-filter").includes(filterBy.location)) {
          if (node.classList.contains("u-hide")) {
            node.classList.remove("u-hide");
          }
        } else {
          if (!node.classList.contains("u-hide")) {
            node.classList.add("u-hide");
          }
          numberOfJobsDisplayed--;
        }
```